### ISSUE TITLE
Bug: await and return not allowed together

### ISSUE THREAD

**Issue by mkahvi** [eslint/eslint#17472]
### Environment

Node version: 18.17
npm version: 9.7.2
Local ESLint version: 8.47
Operating System: Win11

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
{
	"parserOptions": {
		"ecmaVersion": "latest",
		"ecmaFeatures": {
			"globalReturn": true,
			"impliedStrict": true
		}
	}
}
```
</details>

```js
async function XYZ() {}
await XYZ(); // eslint-disable-line
return XYZ(); // eslint-disable-line
```
Same also happens with `eslint-disable-next-line` and `eslint-disable`

### What did you expect to happen?

No errors.

### What actually happened?

With source type set to `script`
```
Parsing error: Unexpected token XYZ
```
or with source type set to `module`
```
Parsing error: 'return' outside of function.
```

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiYXN5bmMgZnVuY3Rpb24gWFlaKCkge31cbi8qIGVzbGludC1kaXNhYmxlICovXG4vLyBlc2xpbnQtZGlzYWJsZS1uZXh0LWxpbmVcbmF3YWl0IFhZWigpOyAvLyBlc2xpbnQtZGlzYWJsZS1saW5lXG4vLyBlc2xpbnQtZGlzYWJsZS1uZXh0LWxpbmVcbnJldHVybiBYWVooKTsgLy8gZXNsaW50LWRpc2FibGUtbGluZSIsIm9wdGlvbnMiOnsiZW52Ijp7ImVzNiI6dHJ1ZX0sInJ1bGVzIjp7fSwicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9LCJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJtb2R1bGUifX19

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

The projects I'm working on allows await and return outside of functions, but that should be irrelevant for eslint ignoring me telling it to ignore the lines. Would like to disable those rules entirely, but there's no option for that that I can tell.

**Comment by Rec0iL99** [comment]
Hi @mkahvi, thanks for the issue. Configuration comments such as `/* eslint-disable */` can only be used to disable rule reports and not parsing errors.

Are you using ES module syntax (import/export) in your code?

**Comment by mkahvi** [comment]
> Are you using ES module syntax (import/export) in your code?

I am not.

**Comment by Rec0iL99** [comment]
In that case, you will have to set the `sourceType` as `script`.

My level of understanding in JavaScript is that we can use `await` only in the scope of an `async` function.

Working [example](https://eslint.org/play/#eyJ0ZXh0IjoiYXN5bmMgZnVuY3Rpb24gWFlaKCkge31cblxuYXN5bmMgZnVuY3Rpb24gbWFpbigpIHtcbiAgYXdhaXQgWFlaKCk7IFxufVxuXG4vLyBnbG9iYWxSZXR1cm4gaXMgc2V0IGZvciB0aGlzIHRvIHdvcmtcbnJldHVybiBYWVooKTsiLCJvcHRpb25zIjp7ImVudiI6eyJlczYiOnRydWV9LCJydWxlcyI6e30sInBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7Imdsb2JhbFJldHVybiI6dHJ1ZX0sImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6InNjcmlwdCJ9fX0=)

> The projects I'm working on allows await and return outside of functions

cc @nzakas @mdjermanovic

**Comment by mkahvi** [comment]
I understand the workaround is possible, but it adds unnecessary roundaboutness just to validate linting that is incorrect in my use.

**Comment by nzakas** [comment]
> The projects I'm working on allows await and return outside of functions, but that should be irrelevant for eslint ignoring me telling it to ignore the lines. Would like to disable those rules entirely, but there's no option for that that I can tell.

Inline disable directives only turn off rules, they do not affect parsing. What you are running into here is a parsing error and there is no way to disable that inline. 

We specifically disable `sourceType: module` and `ecmaFeatures.globalReturn` set to `true` because it confuses other parts of ESLint:

https://github.com/eslint/eslint/blob/9e9edf93ecfa0658e8b79e71bc98530ade150081/lib/linter/linter.js#L1564-L1574

Because we don't have an environment to test that supports both modules and global return, we can't guarantee that ESLint will behave correctly. Prior this issue, I'd never heard of anyone needing such parsing options.

(As a note: `globalReturn` is intended for Node.js only and `sourceType: script`. It's a bit of a hack for CommonJS.)

What are you using that allows both async functions and global return?

**Comment by nzakas** [comment]
No response, so closing.

**Comment by mkahvi** [comment]
Ah, missed the reply, but I don't think I can provide more info that would matter for you.

Essentially my code is pulled in from database and used to construct the function body that is executed and awaited on.
